source/routing/raptor_solution_reader.cpp

```diff
@@ -70,6 +75,120 @@ struct Journey {
+        const DateTime begin_pt_dt =
+            request_clockwise ? begin_section.get_in_dt : begin_section.get_out_dt;
+        const DateTime end_pt_dt =
+            request_clockwise ? end_section.get_out_dt : end_section.get_in_dt;
+        const type::StopTime& begin_st =
+            request_clockwise ? *begin_section.get_in_st : *begin_section.get_out_st;
+        const type::StopTime& end_st =
+            request_clockwise ? *end_section.get_out_st : *end_section.get_in_st;
+        const auto begin_sp_idx = SpIdx(*begin_st.journey_pattern_point->stop_point);
+        const auto end_sp_idx = SpIdx(*end_st.journey_pattern_point->stop_point);
+
+        // is_sp_idx(sp_idx)(sp_dur) returns true if sp_idx == sp_dur.first
+        const auto is_sp_idx = [](const SpIdx idx) {
+            return [idx](RAPTOR::vec_stop_point_duration::const_reference sp_dur) {
+                return sp_dur.first == idx;
+            };
+        };
+        const auto begin_sp_dur_it = boost::find_if(reader.sp_dur_begins, is_sp_idx(begin_sp_idx));
+        assert(begin_sp_dur_it != reader.sp_dur_begins.end());
+        const auto end_sp_dur_it = boost::find_if(reader.sp_dur_ends, is_sp_idx(end_sp_idx));
+        assert(end_sp_dur_it != reader.sp_dur_ends.end());
+        const auto begin_sn_dur = begin_sp_dur_it->second;
+        const auto end_sn_dur = end_sp_dur_it->second;
+
+        // setting objectives
+        begin_dt = begin_pt_dt
+            + (request_clockwise ? -begin_sn_dur.total_seconds() : begin_sn_dur.total_seconds());
+        end_dt = end_pt_dt
+            + (request_clockwise ? end_sn_dur.total_seconds() : -end_sn_dur.total_seconds());
+        sn_dur = begin_sn_dur + end_sn_dur;
+        nb_vj_extentions = count_vj_extentions();
+
+        if (sections.size() > 1) {
+            const auto& data = *reader.raptor.data.pt_data;
+            const auto first_transfer_waiting = get_transfer_waiting(data, sections[0], sections[1]);
+            transfer_dur = first_transfer_waiting.first;
+            min_waiting_dur = first_transfer_waiting.second;
+            const auto* prev = &sections[1];
+            for (auto it = sections.begin() + 2; it != sections.end(); prev = &*it, ++it) {
+                const auto cur_transfer_waiting = get_transfer_waiting(data, *prev, *it);
+                transfer_dur += cur_transfer_waiting.first;
+                min_waiting_dur = std::min(min_waiting_dur, cur_transfer_waiting.second);
+            }
+        }
+    }
+    uint8_t count_vj_extentions() const {
+        uint8_t nb_ext = 0;
+        for (const auto& s: sections) {
+            for (auto* vj = s.get_in_st->vehicle_journey; vj != nullptr; vj = vj->next_vj) {
+                const auto search = boost::find_if(vj->stop_time_list, [&](const type::StopTime& st) {
+                        return &st == s.get_out_st;
+                    });
+                if (search != vj->stop_time_list.end()) { break; }
```

Comment:
shouldn't you check (at least in debug) that the break is used ?
vj extension has been quite troublesome up to now